Incompletely defining a `TypeAliasType` causes mypy to crash

**Crash Report, Reproduction, & Traceback**

See [mypy Playground](https://mypy-play.net/?mypy=latest&python=3.12&flags=show-traceback&gist=c691c554d92d94d2414b45e59254efb2). This causes usability issues with the mypy daemon, as the session would need to be relaunched with a long startup wait.

```python
from typing_extensions import TypeAliasType

MyType = TypeAliasType("MyType")
```
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "mypy/semanal.py", line 7113, in accept
  File "mypy/nodes.py", line 1351, in accept
  File "mypy/semanal.py", line 3119, in visit_assignment_stmt
  File "mypy/semanal.py", line 3863, in check_and_set_up_type_alias
IndexError: list index out of range

main.py:3: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.13.0
main.py:3: : note: use --pdb to drop into pdb
```

**Your Environment**

- Mypy version used: 1.13
- Mypy command-line flags: `--show-traceback`
- Mypy configuration options from `mypy.ini` (and other config files): None

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -4157,8 +4157,12 @@ class SemanticAnalyzer(
             names.append("typing.TypeAliasType")
         if not refers_to_fullname(rvalue.callee, tuple(names)):
             return False
+        if not self.check_typevarlike_name(rvalue, name, rvalue):
+            return False
+        if rvalue.arg_kinds.count(ARG_POS) != 2:
+            return False
 
-        return self.check_typevarlike_name(rvalue, name, rvalue)
+        return True
 
     def analyze_type_alias_type_params(
         self, rvalue: CallExpr

--- a/test-data/unit/check-type-aliases.test
+++ b/test-data/unit/check-type-aliases.test
@@ -1105,6 +1105,10 @@ t1: T1  # E: Variable "__main__.T1" is not valid as a type \
 T3 = TypeAliasType("T3", -1)  # E: Invalid type: try using Literal[-1] instead?
 t3: T3
 reveal_type(t3)  # N: Revealed type is "Any"
+
+T4 = TypeAliasType("T4")  # E: Missing positional argument "value" in call to "TypeAliasType"
+T5 = TypeAliasType("T5", int, str)  # E: Too many positional arguments for "TypeAliasType" \
+                                    # E: Argument 3 to "TypeAliasType" has incompatible type "Type[str]"; expected "Tuple[Union[TypeVar?, ParamSpec?, TypeVarTuple?], ...]"
 [builtins fixtures/tuple.pyi]
 [typing fixtures/typing-full.pyi]
 

